jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1957,6 +1957,15 @@ static void loadLibrary(Class<?> caller, String name, boolean fullPath) {
 		loadLibraryWithClassLoader(name, caller.getClassLoaderImpl());
 }
 
+/*[IF Java15]*/
+static void loadLibrary(Class<?> caller, File file) {
+	loadLibraryWithPath(file.getAbsolutePath(), caller.getClassLoaderImpl(), null);
```

Comment:
Perhaps this should call loadLibraryWithClassLoader(), which might fix the UnsatisfiedLinkError.

Reply:
Will look into it, mark this PR WIP.

Reply:
It doesn't appear the `UnsatisfiedLinkError` during the build process is related to this PR.
The two helper methods added were not invoked before this error.
In addition, the stacktrace doesn't show `java.lang.ClassLoader` or `java.lang.Runtime` which is the caller of these two new methods.

Reply:
It seems there is a newly introduced `jdk.internal.loader.NativeLibraries` which doesn't work well with OpenJ9 `ClassLoader`.

Reply:
Before this `jdk.internal.loader.NativeLibraries` `JDK15` update, `System.loadLibrary(library)` was used which calls `ClassLoader.loadLibraryWithClassLoader()`, and eventually invokes `BytecodeInterpreter.hpp.inlClassLoaderLoadLibraryWithPath()` which calls `registerNativeLibrary()`.

After this update, `internal.loader.NativeLibraries` has its own `private static native boolean load(NativeLibraryImpl impl, String name, boolean isBuiltin)` which invokes `JVM_LoadLibrary()` and `JVM_FindLibraryEntry()`. This seems break `OpenJ9` native library loading mechanism. 

Restoring `System.loadLibrary(library)` within `jdk.internal.loader.NativeLibraries` can skip its natives and use `OpenJ9 inlClassLoaderLoadLibraryWithPath()` which fixes the  `UnsatisfiedLinkError`. However this approach requires a `JCL` patch.

@DanHeidinga @pshipton please comment.

Reply:
Updated the PR and its description, please review.